clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -13,7 +13,7 @@ export interface IHttpConnectionOptions {
     httpClient?: HttpClient;
     transport?: TransportType | ITransport;
     logger?: ILogger | LogLevel;
-    accessTokenFactory?: () => string;
+    accessTokenFactory?: () => string|Promise<string>;
```

Comment:
Can't this just be `Promise<string>`?

Reply:
That would be a breaking change though, right? Keeping it optional seems better to me.

Reply:
There are no breaking changes in the until this library ships. We make breaking changes between previews all the time (in fact we already have a bunch).

Reply:
I think it should stay `string | Promise<string>`. We use the `|` syntax all over the place for convenience features like this, and TypeScript lets you await a literal value. It should be spaced out though, like the others.